src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5460,18 +5493,24 @@ bool StorageReplicatedMergeTree::waitForTableReplicaToProcessLogEntry(
       */
 
     bool waiting_itself = replica == replica_name;
+    bool wait_for_inactive = 0 <= wait_for_inactive_timeout;
```

Comment:
The different `0 <= wait_for_inactive_timeout` or `0 < wait_for_inactive_timeout` checks feel super unnatural to me and forced me to read things several times. It's personal preference, but to me `wait_for_inactive_timeout => 0`  is much clearer.